modules/caddyhttp/fileserver/browse.go

```diff
@@ -206,7 +210,10 @@ func (fsrv *FileServer) serveBrowse(fileSystem fs.FS, root, dirPath string, w ht
 }
 
 func (fsrv *FileServer) loadDirectoryContents(ctx context.Context, fileSystem fs.FS, dir fs.ReadDirFile, root, urlPath string, repl *caddy.Replacer) (*browseTemplateContext, error) {
-	files, err := dir.ReadDir(10000) // TODO: this limit should probably be configurable
+	if fsrv.FileLimit != 0 {
+		maxDirLimit = fsrv.FileLimit
+	}
```

Comment:
Don't set the value of a global variable like this. `file_server` may be loaded more than once with different config, so they would clobber eachother. Make sure to use a local variable here to grab the default value, then override it with the configured one if non-zero.

Reply:
Done.